pyproject.toml

```diff
@@ -58,31 +65,69 @@ dynamic = ["version"]
 [project.optional-dependencies]
 dev = [
   "bump2version",
-  "black",
-  "coverage [toml]",
   "hatch",
-  "isort",
-  "mypy==0.981",
+]
+docs = [  # Required for ReadTheDocs
   "myst_parser",
-  "pylint==2.15.6",
-  "pyroma",
-  "prospector[with_pyroma]",
-  "pytest",
-  "pytest-cov",
   "sphinx",
   "sphinx_rtd_theme",
   "sphinx-autoapi",
+  "coverage[toml]",
+]
+bokeh = [
   "bokeh < 3.0.0",
-  "tox",
 ]
-publishing = ["twine", "wheel"]
+
+[tool.hatch.envs.style]
+detached = false  # Mypy requires package and dependency info
+dependencies = [
+  "flake8",
+  "flake8-pyproject",  # Allows flake8 configuration through this file
+  "pyflakes",  # flake8 plugin, checks for errors in code
+  "mccabe",  # flake8 plugin, checks code complexity
+  "pep8-naming",  # flake8 plugin, checks PEP8 naming conventions
+  "black",
+  "isort",
+  "mypy==0.981",
+]
+
+[tool.hatch.envs.style.scripts]
+check = [
+  "flake8 .",
+  "mypy lilio",
+  "black --check --diff .",
+  "isort --check-only --diff .",
```

Comment:
Might it make sense to run linters after formatters instead, so you'll highlight only remaining issues?

Reply:
Nevermind, I misread this part

Reply:
The `check` script is only for checking the formatting.  black and isort are only run for checking there.

If you want to format first, I set up the `fmt` script, will call the `check` after formatting.